sdk/python/feast/infra/online_stores/remote.py

```diff
@@ -0,0 +1,180 @@
+# Copyright 2021 The Feast Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import json
+from datetime import datetime
+
+import requests
+import logging
+from pydantic import StrictStr
+from feast import Entity
+
+from feast import FeatureView, RepoConfig
+from feast.infra.online_stores.online_store import OnlineStore
+from feast.protos.feast.types.EntityKey_pb2 import EntityKey as EntityKeyProto
+from feast.protos.feast.types.Value_pb2 import Value as ValueProto
+from feast.repo_config import FeastConfigBaseModel
+from typing import Any, Callable, Dict, List, Optional, Sequence, Tuple, Literal, Union
+from google.protobuf.json_format import Parse, ParseDict
+from feast.value_type import ValueType
+from feast.type_map import python_values_to_proto_values
+
+
+logger = logging.getLogger(__name__)
+class RemoteOnlineStoreConfig(FeastConfigBaseModel):
+    """Online store config for Redis store"""
+
+    type: Literal["remote"] = "remote"
+    """Online store type selector"""
+
+    path: StrictStr = "http://localhost:6566"
+    """ str: Path to metadata store.
+    If type is 'remote', then this is a URL for registry server """
+
+
+class RemoteOnlineStore(OnlineStore):
+    
+    
+    """
+    remote online store implementation wrapper to communicate with feast online server.      
+    """
+    def online_write_batch(
+            self,
+            config: RepoConfig,
+            table: FeatureView,
+            data: List[
+                Tuple[EntityKeyProto, Dict[str, ValueProto], datetime, Optional[datetime]]
+            ],
+            progress: Optional[Callable[[int], Any]],
+    ) -> None:
+        assert isinstance(config.online_store, RemoteOnlineStoreConfig)
+        config.online_store.__class__ = RemoteOnlineStoreConfig
+        req_body = self._construct_push_api_json_request(data[0], table)
+        headers = {'Content-type': 'application/json', 'Accept': 'text/plain'}     
+        response = requests.post(f'{config.online_store.path}/push', data=json.dumps(req_body), headers=headers)
+        if response.status_code == 200:
+            logger.debug("Able to write the online features from the feature server.")
+        else:
+            error_message = f"Unable to write the data to online store using feature server API. Error_code={response.status_code}, error_message={response.reason}"
+            logger.error(error_message)
+            raise RuntimeError(error_message)
+
+    def online_read(
```

Comment:
I'm realizing we will probably have the same problem with `FeatureService`s here as in offline server discussed [here](https://github.com/RHEcosystemAppEng/feast/pull/3#issuecomment-2127894618). cc @dmartinol 

This is also probably a very chatty protocol with a list of features always transmitted on the wire even if feature service was requested.